src/request.c

```diff
@@ -58,9 +58,36 @@ request_on_signal_response (XdpRequest *object,
   g_list_free_full (connections, g_object_unref);
 }
 
+static gboolean
+handle_close (XdpRequest *object,
+              GDBusMethodInvocation *invocation)
+{
+  Request *request = (Request *)object;
+  g_autoptr(GError) error = NULL;
+
+  REQUEST_AUTOLOCK (request);
+
+  if (request->exported)
+    {
+      if (request->impl_request &&
+          !xdp_impl_request_call_close_sync (request->impl_request, NULL, &error))
+        {
+          g_dbus_method_invocation_return_gerror (invocation, error);
```

Comment:
If we fail to close the backend here we should probably still unexport the request, and consider it dead.
There is not much else for the client to do, its unlikely closing the requests again would help. This is similar to the close() syscall, which closes the fd even on errors.